### ISSUE TITLE
No metadata file

### ISSUE THREAD

**Issue by MAWENTAO95** [MatteoCaldana/AMG-ANN#1]
I generated a dataset and divided it into a training set and a testing set, but the lack of metadata files resulted in training. No information was found in the code regarding the generation of metadata files.

**Comment by MatteoCaldana** [comment]
Dear Ma Wentao,

the metadata files contain the hyperparameters of the neural network and they must be manually tuned by the user.
They are a CSV file in which each row represents a choice of hyperparameters.
I have pushed some examples in the repository, to have some possible guidelines on the format to follow.

Here is a brief explanation of their meaning
- **batch_size**: batch size used during the training
- **epochs**: number of epochs for which the training lasts
- **lr**: initial learning rate
- **patience**: patience of the learning rate scheduler (reduce on plateau)
- **opt**: optimizer to use
- **loss**: loss function to use (eg. mse or mae)
- **norm_mode**: normalization function used for the pooling, defined in `norm_view` at line 133 of `train_ann.py`
- **inputs**: data frame columns to use as inputs of the NN (separated by a semicolumn)
-  **tgt**: data frame column to predict using the NN
- **learn_err**: add to the NN a second output to learn the error (see Eq.(5) of the paper)
- **act**: activation functions of the NN
- **validation_split**: validation split fraction of the data
- **view_size**: the size of the pooling operator (m in the paper)
- **view_type**: features extracted during the pooling, see function `df_to_model_input` at line 222 of `train_ann.py`
- **join_on**: columns on which [join](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html) the pooling and parameters data frames

moreover, you have to specify either 
- **pretrain**: path of a `.h5` model from which start the training 

or the hyperparameters of the neural net architecture

- cnn_type: type of the CNN part, see `cnn_models.py`
- w1: width of first convolution layer
- d1: depth of first convolution layer
- w2: width of second convolution layer
- d2: depth of second convolution layer
- bn: width of the bottleneck part
- w3: width of the dense part
- d3: depth of the dense part
- kernel_size

Let me know if you need any additional help.
